tests/workbenches/notebook-controller/test_spawning.py

```
@@ -0,0 +1,82 @@
+                {"name": NTB_NAME},
+            )
+        ],
+        indirect=True,
+    )
+    def test_create_simple_notebook(
+        self, admin_client, unprivileged_client, users_namespace, users_persistent_volume_claim
+    ):
+        """
+        # description
+        Create simple Notebook with all needed resources and see if Operator creates it properly
+
+        # contact
+        Contact(name="Jakub Stejskal", email="jstejska@redhat.com"),
+
+        # steps
+            1. Step(
+                value="Create namespace for Notebook resources with proper name, labels and annotations",
+                expected="Namespace is created",
+            ),
+            2. Step(value="Create PVC with proper labels and data for Notebook", expected="PVC is created"),
+            3. Step(
+                value="Create Notebook resource with Jupyter Minimal image in pre-defined namespace",
+                expected="Notebook resource is created",
+            ),
+            4. Step(
+                value="Wait for Notebook pods readiness",
+                expected="Notebook pods are up and running, Notebook is in ready state",
+            ),
+        """
+        with step("Create Notebook CR"):
+            notebook_image: str = get_notebook_image(image_name="jupyter-minimal-notebook", image_tag="2024.2")
+            notebook = load_default_notebook(
+                dyn_client=admin_client, namespace=self.NTB_NAMESPACE, name=self.NTB_NAME, image=notebook_image
+            )
+
+        with step("Wait for Notebook pod readiness"):
+            with notebook:
+                pods = Pod.get(
+                    dyn_client=unprivileged_client, namespace=self.NTB_NAMESPACE, label_selector=f"app={self.NTB_NAME}"
+                )
+                assert pods, "The expected notebook pods were not found"
+                for pod in pods:
+                    pod.wait_for_condition(condition=pod.Condition.READY, status=pod.Condition.Status.TRUE)
```

right now you will fail of the first pod in `wait_for_condition`, depends on what you want to test maybe it's better to collect the failed pods and assert once with all of them.

---

thanks for the note; failing fast is what I want